Tests/SentryTests/Networking/SentryHTTPInterceptorTests.swift

```diff
@@ -0,0 +1,256 @@
+        XCTAssertTrue(configuration.protocolClasses?.first === SentryHttpInterceptor.self)
+    }
+    
+    func testSessionConfigurationWithOtherProtocol() {
+        let configuration = URLSessionConfiguration()
+        configuration.protocolClasses = [AlternativeTestProtocol.self]
+        SentryHttpInterceptor.configureSessionConfiguration(configuration)
+        XCTAssertTrue(configuration.protocolClasses?.first === SentryHttpInterceptor.self)
+    }
+    
+    func testCanInitWithRequestHTTP() {
+        let request = URLRequest(url: URL(string: SentryHTTPInterceptorTests.httpUrl)!)
+        XCTAssertTrue(SentryHttpInterceptor.canInit(with: request))
+    }
+    
+    func testCanInitWithRequestHTTPS() {
+        let request = URLRequest(url: URL(string: SentryHTTPInterceptorTests.httpsUrl)!)
+        XCTAssertTrue(SentryHttpInterceptor.canInit(with: request))
+    }
+    
+    func testCannotInitWithRequestWS() {
+        let request = URLRequest(url: URL(string: SentryHTTPInterceptorTests.wsUrl)!)
+        XCTAssertFalse(SentryHttpInterceptor.canInit(with: request))
+    }
+    
+    func testCannotInitWithSentryRequest() {
+        let request = URLRequest(url: URL(string: fixture.options.dsn!)!)
+        XCTAssertFalse(SentryHttpInterceptor.canInit(with: request))
+    }
+    
+    func testCannotInitWithoutScopeTransaction() {
+        fixture.scope.span = nil
+        let request = URLRequest(url: URL(string: SentryHTTPInterceptorTests.httpsUrl)!)
+        XCTAssertFalse(SentryHttpInterceptor.canInit(with: request))
+    }
+    
+    func testCanonicalRequestForRequest() {
+        let request = URLRequest(url: URL(string: SentryHTTPInterceptorTests.httpUrl)!)
+        let newRequest = SentryHttpInterceptor.canonicalRequest(for: request)
+        
+        let spanHeader = fixture.span.toTraceHeader()
+        XCTAssertNotEqual(request, newRequest)
+        XCTAssertEqual(spanHeader.value(), newRequest.value(forHTTPHeaderField: SENTRY_TRACE_HEADER))
+        
+        let interceptFlag = URLProtocol.property(forKey: SENTRY_INTERCEPTED_REQUEST, in: newRequest) as? NSNumber
+        
+        XCTAssertTrue(interceptFlag!.boolValue)
+    }
+    
+    func testCanonicalRequestForRequestWithoutTransaction() {
+        fixture.scope.span = nil
+        let request = URLRequest(url: URL(string: SentryHTTPInterceptorTests.httpUrl)!)
+        let newRequest = SentryHttpInterceptor.canonicalRequest(for: request)
+        
+        XCTAssertEqual(request, newRequest)
+        XCTAssertNil(newRequest.value(forHTTPHeaderField: SENTRY_TRACE_HEADER))
+    }
+    
+    func testIgnoreAlreadyInterceptedRequest() {
+        let request = URLRequest(url: URL(string: SentryHTTPInterceptorTests.httpUrl)!)
+        let newRequest = SentryHttpInterceptor.canonicalRequest(for: request)
+        XCTAssertFalse(SentryHttpInterceptor.canInit(with: newRequest))
+    }
+    
+    func testProtocolGetRegisteredAndUnregistered() {
```

Comment:
`l`: I think we can move this and `testProtocolDontGetRegistered` to an extra test class `SentryNetworkTrackingIntegrationTests`.